netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1570,7 +1570,8 @@ net_ebpf_extension_sock_addr_authorize_recv_accept_classify(
         goto Exit;
     }
 
-    program_result = net_ebpf_extension_hook_invoke_programs(sock_addr_ctx, &filter_context->base, &result);
+    program_result =
+        net_ebpf_extension_hook_expand_stack_and_invoke_programs(sock_addr_ctx, &filter_context->base, &result);
```

Comment:
I would suggest that `net_ebpf_extension_hook_invoke_programs` takes a new `flags` bitmask parameter. One of the bits can control if it should internally call expand stack. Only sock_addr should pass TRUE in that call. No need in other providers.

Reply:
Available stack expansion api KeExpandKernelStackAndCalloutEx needs a callout. Hence taking the existing approach, which uses KeExpandKernelStackAndCalloutEx with a callout to invoke net_ebpf_extension_hook_invoke_programs.

Reply:
OK - after we discussed this more, please add these to bind and sock_ops as well.